lib/stormpath-sdk/resource/application.rb

```diff
@@ -98,10 +114,21 @@ def get_provider_account request
     Stormpath::Provider::AccountResolver.new(data_store).resolve_provider_account(href, request)
   end
 
+  def oauth_authenticate(options = {})
```

Comment:
“specify actual request objects, instead of just an options array; this would be more object oriented”
But hashes and arrays are also objects in ruby, we don’t have primitive data types ( just kidding :p )

I agree with you, before making changes what do you think of this

```ruby
pasword_grant = Stormpath::Oauth::PasswordGrantRequest.new username, password
result = application.authenticate_oauth(password_grant)
```

regarding the name of the method we already have authenticate_account method which is the main/mostly used method for authentication so in my opinion it can be confusing if we have a method which is just called authenticate on application. I would rename it to authenticate_oauth.  

the other option is to expose to the user the authenticator class itself, in this case I agree on the authenticate name

```ruby
password_grant = Stormpath::Oauth::PasswordGrantRequest.new username, password
auth = Stormpath::Oauth::PasswordGrantAuthenticator.new(application)
result = auth.authenticate(password_grant)
```
————
Unrelated to the task it is common practices in ruby to pass data with blocks(closures). Nothing wrong with using plain ruby objects though.

first two example that came to my mind
https://github.com/octokit/octokit.rb#making-requests
https://github.com/plataformatec/simple_form#simple-fields-for
http://radar.oreilly.com/2014/02/why-ruby-blocks-exist.html